src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/MultiplicativeOperationIterator.java

```diff
@@ -68,7 +68,10 @@ public void open(DynamicContext context) {
                         this.result = new IntegerItem(l * r, ItemMetadata.fromIteratorMetadata(getMetadata()));
                         break;
                     case DIV:
-                        this.result = new IntegerItem(l / r, ItemMetadata.fromIteratorMetadata(getMetadata()));
+                        BigDecimal decLeft = Item.<BigDecimal>getNumericValue(left, BigDecimal.class);
```

Comment:
This will change the return type to decimal even for normal integer division. 4/2 will return 2.0 which is counter intuitive. Maybe we should check if the integer division returns a 0 remain and if it does, don't convert it to a big decimal.